components/arm/xarm/xarm.go

```diff
@@ -152,6 +153,30 @@ func NewxArm(ctx context.Context, r robot.Robot, cfg config.Component, logger go
 	return &xA, nil
 }
 
+// UpdateAction helps hinting the reconfiguration process on what strategy to use given a modified config.
+// See config.UpdateActionType for more information.
+func (x *xArm) UpdateAction(c *config.Component) config.UpdateActionType {
+	remoteAddr := x.conn.RemoteAddr().String()
+	currentHost := string([]rune(remoteAddr)[:len(remoteAddr)-4])
+	if newCfg, ok := c.ConvertedAttributes.(*AttrConfig); ok {
+		if currentHost != newCfg.Host {
+			return config.Reconfigure
+		}
+		if newCfg.Speed == 0 {
```

Comment:
Not sure I like this behavior. Speed being zero would mean that it wasn't passed in and got the default value; I think in this case the correct behavior would be "do not change the speed". Same for acceleration below.

Reply:
makes sense. pushing shortly